### PR TITLE
Pick default OCI Runtime from containers.conf

### PR DESCRIPTION
**Pull request by rhatdan**
Currently we have a weird situation where the user sets the default
runtime in his containers.conf for podman but Buildah is still falling
back to use runc because it was hard coded as the default for Buildah.

I would like to remove this default, but that would theoretically break
the API promise of Buildah.

This should fix https://github.com/containers/podman/issues/8893

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

#### What type of PR is this?

> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test 
> /kind feature
> /kind flake
> /kind other

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL

File: util/types.go
Comment (vrothberg) [inline util/types.go:6]: I think this should be "runc" for now. It's safe for Fedora but who knows in which packages this ends up.

File: util/util.go
Comment (vrothberg) [inline util/util.go:268]:
```suggestion
		logrus.Debugf("Error loading container config when searching for local runtime: %v", err)
```
I wonder whether this error should be fatal.  I didn't check the code but I would assume that we don't get an error if the config does not exist ... but if there's some real config/syntax/permission error here, then it would be very hard to figure out.

File: util/util_test.go
Comment (eriksjolund) [inline util/util_test.go:48]: Maybe this is better?
```	
if runtime != defaultRuntime {
	t.Fatalf("expected %v, got %v", runtime, defaultRuntime)
}
```

File: util/types.go
Comment (rhatdan) [reply util/types.go:6]: I actually think we should remove this altogether or maybe return "".  This field should never be used.  Basically it would only ever be used if containers/common had a serious issue, where their was a coding mistake in the code, which has already gone through  multiple layers of testing.

Comment (rhatdan) [reply util/types.go]: But to get it through, I changed it back. Also fixed the message if containers.conf is screwed up to Warnf level.

File: util/types.go
Comment (saschagrunert) [inline util/types.go:6]:
```suggestion
	// Deprecated: Default runtime should come from containers.conf
	DefaultRuntime = "runc"
```
Note that this change may trigger linters to complain.

File: util/util.go
Comment (saschagrunert) [inline util/util.go:268]: nit:
```suggestion
		logrus.Warnf("Error loading container config when searching for local runtime: %v", err)
```

**Review by vrothberg — COMMENTED** [review]
Can you squash the commits?

**Comment by rhatdan** [comment]
Everything commented on was merged and fixed.

**Review by saschagrunert — APPROVED** [review]
LGTM

**Review by vrothberg — COMMENTED** [review]
/lgtm
/hold

**Comment by vrothberg** [comment]
@rhatdan needs a rebase :(

**Comment by TomSweeneyRedHat** [comment]
LGTM
@vrothberg or @nalind, want to add a final lgtm?

**Pull request merged**
